Add content around address formatting and validation (#822)
BitcoinDesign/Guide: guide/glossary/address.md

</div>

## Visual formatting

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/glossary/address/address-expanded.png"
   retina = "/assets/images/guide/glossary/address/address-expanded@2x.png"
   width = 250
   height = 541
   caption = "Visual formatting can make it easier to compare addresses character by character."
   alt-text = "Screen showing a modal with a bitcoin address in large type."
   layout = "float-right-desktop -background -shadow"
%}

Handling addresses can be stressful when sending bitcoin. Transactions cannot be reversed, and sending to an incorrect address may mean loss of funds. While address formats are what they are, visual formatting can make it easier for users to compare addresses and ensure their accuracy.

Especially when addresses are displayed in a compacted way, make sure to offer a more easily readable alternative.

</div>

## Address validation

> Validation with visual feedback can reduce user anxiety about making mistakes. While addresses look like randomly assembled characters, there are patterns that can be used to identify whether an address is valid or not. It may be possible to even identify which characters are incorrect, based on the format. There are limits to this, as too many mistakes may result in a valid address, except it is a different one that the user intended to enter.

Reviewer: ```suggestion
Validation with visual feedback can reduce user anxiety about making mistakes. While addresses look like randomly assembled characters, there are patterns that can be used to identify whether an address is valid or not. For [segwit]({{'/guide/glossary/address/#segwit-address---p2wpkh' | relative_url}}) and [taproot]({{'/guide/glossary/address/#taproot-address---p2tr' | relative_url}}) addresses, it may be possible to identify which characters are incorrect. However, there are limits to this. Too many mistakes may result in an address that is valid but different from what the user intended to enter.
```
For the advanced error detection, I think you're trying to describe bech32 encoded addresses, which for our purposes is basically segwit and taproot addresses. Updated text. Also included style suggestions.

Author: Thanks for clarifying that. I could not find any info on exactly what's possible for validation.
